### ISSUE TITLE
[a11y] 9.1.3.1h - Labels of form elements programatically determinable

### ISSUE THREAD

**Issue by tbsbdr** [owncloud/web#5382]
- Context: The Web UI should be accessible according to the WCAG2.1 /BITV 2.0 guidelines. The number in the title relates to the test-steps: https://www.bitvtest.de/bitv_test/das_testverfahren_im_detail/pruefschritte.html
- "Status" relates to the evaluation scheme https://www.bitvtest.de/bitv_test/das_testverfahren_im_detail/verfahren.html#c1262 
- "a11y" is short for accessibility
- ----------------

## General notes

- [x] "Personen hinzufügen" ("Öfentlicher Link") / Dropdown element "Rolle": Despite the visual label "Rolle" is correctly implemented as label-element and also programatically connected with for and id attributes, it will not be read by screen readers when focussing the dropdown. This is because the way the dropdown element was implemented with ARIA roles / attributes. With current implementation the text „nach Option suchen“ will be read out twice instead of reading the text „Rolle“ - probably because `aria-label` and `aria-labelledby` are used the same time. See also 9.4.1.2.

- [ ] "Öffentlicher Link" / Element "Ablaufdatum": Despite the visual label "Ablaufdatum" being correctly implemented and also programatically connected with `for` and `id` attributes, it will not be read by screen readers when focussing the element.

## Status: partly fulfilled
- Page 1 (Alle Dateien): Fixed by issue https://github.com/owncloud/web/issues/5394
- Page 2 (Mit anderen geteilten Dateien)

## Status: fulfilled
- Page 3 (Gelöschte Dateien)

## Status: not applicable
- Page 4 (Konto)

**Comment by paulcod3** [comment]
Point 1 is already beeing fixed by issue https://github.com/owncloud/web/issues/5394
Point 2 is really weird and Im not yet sure what the right solution is, when I put the input element of the datepicker right under the label in DOM it seems to read the label out fine.
